spec/ng2/async_spec.js

```diff
@@ -54,4 +54,20 @@ describe('async angular2 application', function() {
 
     expect(timeout.$('.val').getText()).toEqual('10');
   });
+
+  it('should wait for a series of periodic increments', function() {
+    var timeout = $('#periodicIncrement');
+    timeout.$('.action').click();
+
+    // Waits for the val to count to 1 and 2.
+    var EC = protractor.ExpectedConditions;
+    browser.wait(EC.textToBePresentInElement(timeout.$('.val'), '1'), 2000);
+    expect(timeout.$('.val').getText()).toEqual('1');
+    browser.wait(EC.textToBePresentInElement(timeout.$('.val'), '2'), 2000);
+    expect(timeout.$('.val').getText()).toEqual('2');
+
+    timeout.$('.cancel').click();
```

Comment:
Race condition here if the number gets updated before the click on the cancel button gets through